assets/app.js

```diff
@@ -794,39 +788,88 @@ new Vue({
         },
     },
     methods: {
-        validateRejectedProduct: function (type, product) {
-            if (type == 'materials' && product['is_rejected'] && product['rejected_' + type]) {
-                if (!(product['is_rejected'] && product['rejected_' + type])) {
-                    return true
+        validateCacheMaterial: function () {
+            active_languages = this.activeLanguagesIds
+            this.products.forEach(function (product) {
```

Comment:
What about something like:
```
 this.products.forEach(function (product) {
    active_languages.forEach(function (language) {
       // we assume that deleted materials are already removed from product.materials
        current_materials[language] = product.materials[language].join(" / ")
        if(current_materials[language]===product.cached_materials[language]){
             product.cached_materials[language]['is_overridden'] = false
        }
        else{
            product.cached_materials[language]['is_overridden'] = false
        }
    }
}
```

it should do pretty much the same thing or not?

Reply:
the inner loop, why not: `product.cached_materials[language]['is_overridden'] = !current_materials[language]===product.cached_materials[language]` :slightly_smiling_face: